File: generate_primes.py
from time import time
import sys

def is_prime(n):
    if n <= 1: 
        return False
    if n <= 3:
        return True
    if n % 2 == 0 or n % 3 == 0:
        return False
    i = 5
    while i * i <= n:
        if n % i == 0 or n % (i + 2) == 0:
            return False
        i += 6
    return True

def generate_primes(filename, limit):
    primes = []
    num = 2
    count = 0
    while count < limit:
            if is_prime(num):
                primes.append(str(num))
                count += 1
            num += 1

    with open(filename, 'w') as file:
        file.write("\n".join(primes))
        

# Default limit
limit = 10000

# Check if an argument is provided
if len(sys.argv) > 1:
    try:
        input_limit = int(sys.argv[1])
        limit = input_limit
    except ValueError:
        print("Invalid number provided, using default limit of 10000")

start_time = time()
generate_primes("primes_python.txt", limit)
end_time = time()

print(f"Time taken: {end_time - start_time} seconds")

File: generate_primes.js
const fs = require("fs");

function isPrime(num) {
  if (num <= 1) return false;
  if (num <= 3) return true;

  if (num % 2 === 0 || num % 3 === 0) return false;

  for (let i = 5; i * i <= num; i += 6) {
    if (num % i === 0 || num % (i + 2) === 0) return false;
  }
  return true;
}

function generatePrimes(filename, limit) {
  let num = 2;
  let count = 0;
  const stream = fs.createWriteStream(filename);

  while (count < limit) {
    if (isPrime(num)) {
      stream.write(num + "\n");
      count++;
    }
    num++;
  }

  stream.end();
}

let limit = 10000;

// Check if an argument is provided
if (process.argv.length > 2) {
  const inputLimit = parseInt(process.argv[2], 10);
  if (!isNaN(inputLimit)) {
    limit = inputLimit;
  } else {
    console.log("Invalid number provided, using default limit of 10000");
  }
}

const startTime = new Date();
generatePrimes("primes_node.txt", limit);
const endTime = new Date();

console.log(`Time taken: ${(endTime - startTime) / 1000} seconds`);
